src/Core/Managed/Shared/Channel/InMemoryTransmitter.cs

```diff
@@ -156,13 +154,14 @@ private void Dispose(bool disposing)
         {
             if (Interlocked.Increment(ref this.disposeCount) == 1)
             {
-                // Stops the runner.
+                // Stops the runner loop.
                 this.enabled = false;
 
                 if (this.startRunnerEvent != null)
                 {
-                    // call Set to to prevent waiting for the next interval. 
+                    // Call Set to prevent waiting for the next interval in the runner, and then Dispose to prevent a race condition with dispose vs. the wait timeout.
                     this.startRunnerEvent.Set();
+                    this.startRunnerEvent.Dispose();
```

Comment:
this is still risky... because if this Dispose() is called before the Runner gets a chance to finish WaitOne, then WaitOne() call in Runner can get ObjectDisposedException.